package collector;

message JobMeta {
	enum JobStatus {
		NOT_STARTED = 0;
		IN_PROGRESS = 1;
		JB_COMPLETE = 2;
		ERR_OCCURED = 3;
	}
	
	required string name	= 1;
	required float	time	= 2	[default = 0];
	required uint32 fails	= 3 [default = 0];

	/*optional uint32	id		= 2	[default = 0];
	optional uint32	type	= 3	[default = 0];
	optional uint32 node	= 4 [default = 0];
	optional uint32 worker	= 5 [default = 0];
	optional JobStatus status = 7 [default = NOT_STARTED];*/
}

message Job {
	required JobMeta meta = 1;
}

message JobExample {
	required JobMeta meta = 1;
	optional string text = 2;
}

message JobOrmReaderRange {
	required JobMeta meta = 1;
	optional uint32 size = 2;					/* (from) start page*/
}

/* raw social response */
message DataRSR {
	optional string fb = 1;						/* Facebook*/
	optional string tw = 2;						/* Twitter*/
	optional string tp = 3;						/* Topsy*/
	optional string su = 4;						/* StumbleUpon*/
	optional string dg = 5;						/* Digg*/
	optional string gp = 6;						/* GooglePlus*/
	optional string li = 7;						/* LinkedIn*/
	optional string sc = 8;						/* SocialCounter*/
}

/* social response */
message DataSR {
	optional int32 fb = 1;						/* Facebook*/
	optional int32 tw = 2;						/* Twitter*/
	optional int32 t1 = 3;						/* Topsy All*/
	optional int32 t2 = 4;						/* Topsy Influential*/
	optional int32 su = 5;						/* StumbleUpon*/
	optional int32 dg = 6;						/* Digg*/
	optional int32 gp = 7;						/* GooglePlus*/
	optional int32 li = 8;						/* LinkedIn*/
}